common/src/api/internal/nexus.rs

```diff
@@ -108,6 +109,80 @@ pub struct SledInstanceState {
 
     /// The most recent state of the sled's VMM process.
     pub vmm_state: VmmRuntimeState,
+
+    /// The current state of any in-progress migration for this instance, as
+    /// understood by this sled.
+    pub migration_state: Option<MigrationRuntimeState>,
+}
+
+/// An update from a sled regarding the state of a migration, indicating the
+/// role of the VMM whose migration state was updated.
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema)]
+pub struct MigrationRuntimeState {
+    pub migration_id: Uuid,
+    pub state: MigrationState,
+    pub role: MigrationRole,
+    pub gen: Generation,
+
+    /// Timestamp for the migration state update.
+    // TODO(eliza): could this just be the VMM state timestamp?
```

Comment:
I think I ought to remove this comment. If we intend to use these timestamps for tracking the duration of migrations, we probably want to separate them from the "time state updated" timestamp.
```suggestion
```